overlord/state/taskrunner.go

```diff
@@ -177,6 +198,41 @@ func (r *TaskRunner) run(t *Task) {
 	})
 }
 
+func (r *TaskRunner) clean(t *Task) {
+	select {
+	case <-t.Change().Ready():
+	default:
+		return
```

Comment:
I would add a comment here:

// whole Change is not ready, don't run cleanups yet